code/modules/mob/living/silicon/ai/ai.dm

```diff
@@ -636,64 +638,43 @@ var/list/ai_verbs_default = list(
-
-		var/personnel_list[] = list()
-		var/current_mobs = list()
-
-		for(var/mob/living/carbon/human/H in human_mob_list)
-			current_mobs[H.real_name] = H
-		for(var/datum/record/general/t in SSrecords.records_locked)//Look in data core locked.
-			personnel_list["[t.name]: [t.rank]"] = t.photo_front //Pull names, rank, and image.
-			if(current_mobs[t.name])
-				personnel_list["[t.name]: [t.rank]"] = list("mob" = current_mobs[t.name], "image" = t.photo_front)
-
-		if(personnel_list.len)
-			input = input("Select a crew member:") as null|anything in personnel_list
-			var/selection = personnel_list[input]
-			var/icon/character_icon
-			if(selection && istype(selection, /list))
-				var/mob/living/carbon/human/H = selection["mob"]
-				if (H.near_camera())
-					character_icon = new('icons/mob/human.dmi', "blank")
-					for(var/renderdir in cardinal)
-						character_icon.Insert(getHologramIcon(getFlatIcon(H, renderdir)), dir = renderdir)
-				else
-					character_icon = getHologramIcon(icon(selection["image"]))
-			if(selection && istype(selection, /icon))
-				character_icon = getHologramIcon(icon(selection))
-			if(character_icon)
-				qdel(holo_icon) // Clear old icon so we're not storing it in memory.
-				holo_icon = character_icon
+	if(alert(usr, "Would you like to select a hologram based on a humanoids within camera view or switch to a unique avatar?",,"Humanoids","Unique") == "Humanoids")
+		var/list/selectable_humans = list()
+		for(var/thing in human_mob_list)
+			var/mob/living/carbon/human/H = thing
+			if(H.near_camera())
+				selectable_humans[H.name] = H
+		if(length(selectable_humans))
+			var/chosen_human = input(usr, "Select the humanoid whose form you wish to emulate.", "Hologram Select") as null|anything in selectable_humans
+			if(!chosen_human)
+				return
+			var/mob/living/carbon/human/H = selectable_humans[chosen_human]
+			holo_icon.appearance = H.appearance
 		else
-			alert("No suitable records found. Aborting.")
-
+			to_chat(usr, SPAN_WARNING("There are no humanoids within camera view to base your hologram on."))
 	else
-		var/icon_list[] = list(
-		"default",
-		"floating face",
-		"carp",
-		"custom"
-		)
-		input = input("Please select a hologram:") as null|anything in icon_list
+		input = input("Please select a hologram:") as null|anything in list("default", "floating face", "carp", "custom")
 		if(input)
-			qdel(holo_icon)
 			switch(input)
 				if("default")
-					holo_icon = getHologramIcon(icon('icons/mob/AI.dmi',"holo1"))
+					set_hologram_unique(icon('icons/mob/AI.dmi', "holo1"))
 				if("floating face")
-					holo_icon = getHologramIcon(icon('icons/mob/AI.dmi',"holo2"))
+					set_hologram_unique(icon('icons/mob/AI.dmi', "holo2"))
 				if("carp")
-					holo_icon = getHologramIcon(icon('icons/mob/AI.dmi',"holo4"))
+					set_hologram_unique(icon('icons/mob/AI.dmi', "holo4"))
```

Comment:
Would you mind just changing the hologram icon state names and doing :
`set_hologram_unique(icon('icons/mob/AI.dmi', input)` instead of this dumb switch and only overriding on custom sprites?